core/src/main/java/org/infinispan/factories/InternalCacheFactory.java

```diff
@@ -52,6 +73,39 @@
       }
    }
 
+   private boolean allowSimpleCache(Configuration configuration) {
+      if (!configuration.allowSimple()) {
+         return false;
+      }
+      ClusteringConfiguration clustering = configuration.clustering();
+      TransactionConfiguration transaction = configuration.transaction();
+      CustomInterceptorsConfiguration interceptors = configuration.customInterceptors();
+      PersistenceConfiguration persistence = configuration.persistence();
+      InvocationBatchingConfiguration invocationBatching = configuration.invocationBatching();
+      IndexingConfiguration indexing = configuration.indexing();
+      StoreAsBinaryConfiguration storeAsBinary = configuration.storeAsBinary();
+      JMXStatisticsConfiguration jmxStatistics = configuration.jmxStatistics();
+      // TODO: eventually support transactional caches as well
+      if (clustering != null && clustering.cacheMode().isClustered()) {
```

Comment:
These don't need to be a bunch of if elses since you are returning in the if.  It seems this would be cleaner if it was just a local variable assignment and then use a single appropriately for that variable.

```
ClusteringConfiguration clustering = configuration.clustering();
if (clustering != null && clustering.cacheMode().isClustered()) {
   return false;
}
TransactionConfiguration transaction = configuration.transaction();
if (transaction != null && transaction.transactionMode().isTransactional()) {
   return false;
}
... etc ...
```